### PR TITLE
Remove _NoLogging methods from VM 

### LINKED ISSUE #89634: _NoLogging methods in VM
Since #68717 there are a bunch of `Xyz` and `Xyz_NoLogging` method pairs in VM code.
In most cases `Xyz` simply calls the _NoLogging variant or has the same implementation, e.g.
https://github.com/dotnet/runtime/blob/f5ca995bb6c9ab79993c3bd63a2391b393c4fd14/src/coreclr/vm/methodtable.h#L879-L890

I think the only methods that differ in behaviour are `MethodTable::GetClass_NoLogging`/`MethodTable::GetClass`.

Could these _NoLogging variants be removed or are they kept for some reason?

### PR DESCRIPTION
**Pull request by jasper-d**
Removes all _NoLogging methods.

Fixes #89634

### CODE REVIEW

**Comment by jkotas** [comment]
>I think that assert wouldn't be a problem except for GetClassWithPossibleAV. Also, GetClass_NoLogging has FORCEINLINE while GetClass has only inline. Not sure what the best course of action here is.

- Rename `GetClass_NoLogging` to `GetClassWithPossibleAV`
- Change `GetClass` to call it `GetClassWithPossibleAV` and assert that the result is non-null. Mark both with `FORCEINLINE` for a good measure.

**Comment by jasper-d** [comment]
> * Rename `GetClass_NoLogging` to `GetClassWithPossibleAV`

@jkotas There is already `GetClassWithPossibleAV` but that has `CANNOT_HAVE_CONTRACT`:
https://github.com/dotnet/runtime/blob/f6a66fafa043953284ee7ca823df4e5219d57640/src/coreclr/vm/methodtable.inl#L63-L68

So rename `GetClassWithPossibleAV` to `GetClassWithPossibleAVNoContract` and `GetClass_NoLogging` to `GetClassWithPossibleAV`?

**Comment by jkotas** [comment]
I would delete the existing `GetClassWithPossibleAV` and move the comment from the existing `GetClassWithPossibleAV` to the renamed `GetClass_NoLogging`. 

The extra `CANNOT_HAVE_CONTRACT` annotation in the existing `GetClassWithPossibleAV` is not important.

File: src/coreclr/vm/methodtable.h [hunk 1]
```diff
@@ -1821,9 +1807,7 @@ class MethodTable
 
     PTR_EEClass GetClass();
 
-    inline PTR_EEClass GetClass_NoLogging();
-
-    PTR_EEClass GetClassWithPossibleAV();
+    inline PTR_EEClass GetClassWithPossibleAV();
```
Comment (jkotas) [inline src/coreclr/vm/methodtable.h:1810]:
```suggestion
    PTR_EEClass GetClassWithPossibleAV();
```

**Review by jkotas — APPROVED** [review]
Thank you!

**Comment by jasper-d** [comment]
Thank you for the help!

**Pull request merged**